### PR TITLE
chore(financial-aid): refactor custom components out

### PR DESCRIPTION
**Pull request by jonnigs**
## What

* Refactor out custom components
* Reorganize code in `forms` folder into one file per screen
* Reorganize remaining custom components in the `fields` folder
* Move svg files into the new folder `assets`
* Make folder `components` with `tsx` components that are used by the custom components
* New form components: buildAccordionField and BuildBankAccountField
* Option to have a dynamic logo on applications

## Why

* Minimize the use of custom components, making applications more uniform and maintainable
* Organize code to be more similar to other applications
* Add new shared components so other applications can use them

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Enhanced the spouse financial information section by simplifying data extraction and rendering logic.
  - Introduced new utility functions for structured data retrieval related to spouse and summary information.
  - Added a multi-field component for a confirmation section in the financial aid application.
  - Introduced a section for managing spouse tax return file uploads.
  - Added new types and field builder functions for accordion and bank account fields.

- **Bug Fixes**
  - Updated the `MissingFilesConfirmation` component to improve prop handling and output clarity.

- **Refactor**
  - Refactored multiple components for improved structure and readability, including the `SummaryForm` and `SpouseSummaryForm`.
  - Streamlined export statements to enhance component usability.

### CODE REVIEW

**Review by Toti91 — CHANGES_REQUESTED** [review]
Couple of things. Im mainly worried about if we are removing important functionality or not

File: libs/application/templates/financial-aid/src/fields/Summary/MissingFilesConfirmation.tsx [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
+    ? 'missingFilesSpouse'
+    : 'missingFiles'
   const commentType = 'fileUploadComment'
 
   return (
     <>
-      <Text variant="h3" color={'mint600'} marginBottom={3}>
-        {formatMessage(missingFiles.confirmation.subtitle)}
-      </Text>
-
```
[the remark this answers is not in the document]
Comment (jonnigs) [reply libs/application/templates/financial-aid/src/fields/Summary/MissingFilesConfirmation.tsx:23]: This was moved out of the custom component and is now on the section

File: libs/application/templates/financial-aid/src/fields/Summary/FileList.tsx [hunk 2]
```diff
@@ -49,11 +38,8 @@ const FileList = ({ files, applicationSystemId }: Props) => {
               <div className={styles.name}>
                 <Text variant="small">{file.name}</Text>
               </div>
-              <Text variant="small">{`${formatMessage(
-                missingFiles.confirmation.file,
-              )} • ${getFileSizeInKilo(file)} KB`}</Text>
```
[the remark this answers is not in the document]
Comment (jonnigs) [reply libs/application/templates/financial-aid/src/fields/Summary/FileList.tsx:54]: This code was previously unreachable. The size doesn't come with the file object so this probably didn't work to begin with

File: libs/application/templates/financial-aid/src/fields/Summary/FileList.tsx [hunk 3]
```diff
@@ -28,17 +21,13 @@ const FileList = ({ files, applicationSystemId }: Props) => {
...
+          <div
             className={cn({
               [`${styles.filesLink}`]: true,
               [styles.hoverState]: file.id,
             })}
             key={'file-' + i}
-            onClick={(e) => {
-              e.preventDefault()
-              openFileById(file.id as string)
-            }}
```
[the remark this answers is not in the document]
Comment (jonnigs) [reply libs/application/templates/financial-aid/src/fields/Summary/FileList.tsx:41]: Yes, this is a very strange behaviour though. This allows you to open the file that you just uploaded. This is a part of the previously unreachable code.

File: libs/application/templates/financial-aid/src/lib/FinancialAidTemplate.ts [hunk 4]
```diff
@@ -225,7 +222,7 @@ const FinancialAidTemplate: ApplicationTemplate<
             {
               id: Roles.SPOUSE,
               formLoader: () =>
-                import('../forms/SpouseSubmitted').then((module) =>
+                import('../forms/SpouseSubmittedForm/index.ts').then((module) =>
```
Comment (Toti91) [inline libs/application/templates/financial-aid/src/lib/FinancialAidTemplate.ts:225]: `/index.ts` not needed

**Review by MargretFinnboga — APPROVED** [review]
Looks good ✨  
would recommend breaking this into smaller pr's in the future  👩🏼‍🚀

**Comment by jonnigs** [comment]
> 👩🏼‍🚀

Totally agree, got a bit carried away!

**Pull request merged**